File: packages/types/src/order.ts
import type { AgentThought } from './agent';

export type OrderStatus =
  | 'pending'
  | 'processing'
  | 'complete'
  | 'failed'
  | 'interpreting'
  | 'answered';

export interface EvidenceItem {
  type: 'count' | 'comparison' | 'detection' | 'measurement';
  description: string;
  value?: number | string;
}

export interface OrderResult {
  answer: string;
  confidence: 'high' | 'medium' | 'low';
  evidence: EvidenceItem[];
  rawAnalytics?: Record<string, unknown>;
  imageryUrl?: string;
  capturedAt?: string;
}

export interface Order {
  id: string;
  watchId: string;
  skyfiOrderId?: string;
  skyfiArchiveId?: string;
  status: OrderStatus;
  sensorType: string;
  analyticsType?: string;
  costUsd?: number;
  createdAt: string;
  updatedAt: string;
  result?: OrderResult;
  agentThoughts?: AgentThought[];
}

File: apps/api/src/schemas/order.py
from pydantic import BaseModel
from typing import Optional


class OrderResponseSchema(BaseModel):
    id: str
    watch_id: str
    skyfi_order_id: Optional[str]
    status: str
    sensor_type: str
    analytics_type: Optional[str]
    cost_usd: Optional[float]
    created_at: str
    updated_at: str
    answer: Optional[str]
    confidence: Optional[str]
    evidence: Optional[list]
    captured_at: Optional[str]
    agent_thoughts: Optional[list]
